src/main/fc/rc_adjustments.c

```diff
@@ -331,11 +363,18 @@ static void applyStepAdjustment(controlRateConfig_t *controlRateConfig, uint8_t
         blackboxLogInflightAdjustmentEvent(ADJUSTMENT_D_SETPOINT_TRANSITION, newValue);
         break;
     default:
+    newValue = 0;
         break;
     };
+
+#ifdef OSD
+#if (FLASH_SIZE > 2048)
+    osdShowAdjustment(adjustmentLabels[adjustmentFunction].label, newValue);
+#endif
+#endif
 }
 
-static void applySelectAdjustment(uint8_t adjustmentFunction, uint8_t position)
+static void applySelectAdjustment(int adjustmentFunction, uint8_t position)
```

Comment:
Change  `applyStepAdjustment` and `applySelectAdjustment` to return `newValue`, then in `processRcAdjustments` call `osdShowAdjustment` if returned value is non-zero.

Reply:
done